File: structural-patterns/adapter/go/adapter/MessageSenderAdapter.go
package adapter

import (
	"adapter/ourcode/model"
	lib "adapter/thirdpartylib"
	libMsgModel "adapter/thirdpartylib/model"
)

type MessageSenderAdapter struct {
	CoolSender lib.CoolMessageSenderImpl
}

func (adapter *MessageSenderAdapter) SendMessage(msgType model.MessageType, message string) {
	msg := libMsgModel.Message{
		Message:     message,
		MessageType: msgType.String(),
	}
	adapter.CoolSender.SendCoolMessage(msg)
}

File: structural-patterns/adapter/java/src/main/java/org/example/adapter/MessageSenderAdapter.java
package org.example.adapter;

import org.example.ourcode.MessageSender;
import org.example.ourcode.model.MessageType;
import org.example.thirdpartylib.CoolMessageSender;
import org.example.thirdpartylib.model.Message;

public class MessageSenderAdapter implements MessageSender {

    private final CoolMessageSender coolMessageSender;

    public MessageSenderAdapter(CoolMessageSender coolMessageSender) {
        this.coolMessageSender = coolMessageSender;
    }

    @Override
    public void sendMessage(MessageType type, String message) {
        System.out.print("VIA ADAPTER - ");
        coolMessageSender.sendCoolMessage(new Message(message, type.name()));
    }
}
